ui/app/templates/components/namespace-picker.hbs

```diff
@@ -1,119 +1,26 @@
-            >
-              <div class="is-flex-align-baseline">
-                <Hds::Button
-                  @text="Go back"
-                  @icon="chevron-left"
-                  @isIconOnly={{true}}
-                  @color="tertiary"
-                  class="is-flex-align-baseline"
-                />
-                <p class="is-size-8 has-text-grey has-text-weight-semibold is-uppercase">Namespaces</p>
-              </div>
-            </NamespaceLink>
-          {{/if}}
-        </div>
-        {{#if (includes "" this.lastMenuLeaves)}}
-          {{! leaf is '' which is the root namespace, and then we need to iterate the root leaves }}
-          <div class="leaf-panel {{if (eq '' this.currentLeaf) 'leaf-panel-current' 'leaf-panel-left'}} ">
-            {{#each this.rootLeaves as |rootLeaf|}}
-              <NamespaceLink @targetNamespace={{rootLeaf}} @class="namespace-link" @showLastSegment={{true}} />
-            {{/each}}
-          </div>
-        {{/if}}
-        {{#each this.lastMenuLeaves as |leaf|}}
-          {{#if leaf}}
-            <div
-              class="leaf-panel
-                {{if (eq leaf this.currentLeaf) 'leaf-panel-current' 'leaf-panel-left'}}
-                {{if (and this.isAdding (eq leaf this.changedLeaf)) 'leaf-panel-adding'}}
-                {{if (and (not this.isAdding) (eq leaf this.changedLeaf)) 'leaf-panel-exiting'}}
-                "
-            >
-              {{#each-in (get this.namespaceTree leaf) as |leafName|}}
-                <NamespaceLink
-                  @targetNamespace={{concat leaf "/" leafName}}
-                  @class="namespace-link"
-                  @showLastSegment={{true}}
-                />
-              {{/each-in}}
-            </div>
-          {{/if}}
-        {{/each}}
-        {{#if this.canList}}
-          <div class="leaf-panel leaf-panel-current">
-            <div class="level">
-              <span class="level-left">
-                <LinkTo @route="vault.cluster.access.namespaces" class="is-block namespace-link namespace-manage-link">
-                  Manage Namespaces
-                </LinkTo>
-              </span>
-              <span class="level-right">
-                <Hds::Button
-                  @text="Refresh namespace list"
-                  @icon="reload"
-                  @isIconOnly={{true}}
-                  @color="tertiary"
-                  data-test-refresh-namespaces
-                  {{on "click" (action "refreshNamespaceList")}}
-                />
-              </span>
-            </div>
-          </div>
-        {{/if}}
-      </div>
-    </D.Content>
-  </BasicDropdown>
+  <Hds::Form::SuperSelect::Single::Field
+    @ariaLabel="Namespace"
+    @label="Namespace"
+    @onChange={{this.onChange}}
+    @options={{this.options}}
+    @placeholder="Search"
+    @searchEnabled={{true}}
+    @selected={{this.selected}}
+    @selectedItemComponent={{component "namespace-picker/selected-option"}}
```

Comment:
Note: This implementation isn't in the documented hds code examples; instead I referenced the [HDS source code for the examples](https://github.com/hashicorp/design-system/blob/9e7362a8d048f88a361636a2564646ba4b65b77f/showcase/app/templates/components/form/super-select.hbs#L337).

Reply:
Out of curiosity, why is passing this as an arg necessary? Wondering if it's possible to leverage the [rich content options](https://helios.hashicorp.design/components/form/super-select?tab=code#rich-content-options) and just render the component in the `<F.Options>` block. I'm likely missing an implementation detail!
 
```hbs
{{#let F.options as |option|}}
   <NamespacePicker::SelectedOption @option={{option}} / >
 {{/let}}
```

Reply:
So maybe there's a better way to do this, but the code snippet you shared would affect the listed options inside of the dropdown. `@selectedItemComponent` is used to display the selected option, which is displayed when the namespace picker is closed. 

For example, in this screenshot, the `@selectedItemComponent` template is used to display an icon alongside 'grandchild1' instead of displaying 'child1/grandchild1': 

![Screenshot 2025-03-28 at 10 40 24 AM](https://github.com/user-attachments/assets/0f49a080-addb-4f67-af67-d2f9c36fe4f4)

Reply:
Ahh I see - I missed on my initial pass that this was specifically for the selected option. Using the arg makes sense, my initial scrutiny was also because you mentioned pulling it from the source code (and I was concerned about accessibility), but it looks like they do actually mention it in the docs [here](https://helios.hashicorp.design/components/form/super-select?tab=code#selected-item-component)
<img width="500" alt="Screenshot 2025-03-28 at 11 38 03 AM" src="https://github.com/user-attachments/assets/1d4e5377-356e-4ea2-89c8-4ee1364f787e" />